### DISCUSSION TITLE
Failed to connect to generic netlink. socket(PF_PACKET) failed: Permission denied Failed initializing wireless card(s): wlan0

### DISCUSSION THREAD

**Discussion by CarrelloDaSpesa** [aircrack-ng/aircrack-ng#2577] [General]
@aircrack-ng im having trouble using airodump-ng wlan0 
┌──(root㉿localhost)-[~]
└─# airodump-ng wlan0
Failed to connect to generic netlink.
socket(PF_PACKET) failed: Permission denied
Failed initializing wireless card(s): wlan0

I dont know what to do, can you pls help me?

**Comment by Mister-X-** [comment]
1. This is not an issue. It has been converted to a discussion. FYI, if you are filing an issue, provide the information requested, or we'll close it (lacking information to be actionable).
2. You need to put it in monitor mode first

If it's still an issue, provide the output of `airmon-ng --debug`.

**Reply by CarrelloDaSpesa** [reply]
┌──(root㉿localhost)-[~/aircrack-ng]
└─# airmon-ng --debug
This program cannot continue without a working sysfs.
/sys/class is missing

**Reply by Mister-X-** [reply]
There is your problem, there is no sysfs.

**Reply by CarrelloDaSpesa** [reply]
So what do i need to do?

**Reply by Mister-X-** [reply]
I don't know. You'll have to figure out why `/sys/class` is missing and fix that.

**Reply by CarrelloDaSpesa** [reply]
Ok, thank you
